full-moon/src/ast/types.rs

```diff
@@ -526,6 +526,56 @@ impl<'a> TypeSpecifier<'a> {
     }
 }
 
+/// A type argument specified in a callback type, the `count: number` in `(count: number) -> ()`
+#[derive(Clone, Debug, PartialEq, Owned, Node, Visit)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub struct TypeArgument<'a> {
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub(crate) name: Option<(TokenReference<'a>, TokenReference<'a>)>,
+    #[cfg_attr(feature = "serde", serde(borrow))]
+    pub(crate) type_info: TypeInfo<'a>,
+}
+
+impl<'a> TypeArgument<'a> {
+    /// Creates a new TypeArgument with the given type info
+    pub fn new(type_info: TypeInfo<'a>) -> Self {
+        Self {
+            name: None,
+            type_info,
+        }
+    }
+
+    /// The name of the argument split into identifier and punctuation: `count:` in `count: number`.
+    pub fn name(&self) -> Option<&(TokenReference<'a>, TokenReference<'a>)> {
```

Comment:
I forget, are our other tuple results like `Option<&(T1, T2)>`? I thought they were `Option<(&T1, &T2)>`?

Reply:
In Block it seems to be `Option<&(T1, T2)>`

Reply:
Alright, works for me.